frontend/src/hex/HexView.svelte

```diff
@@ -182,7 +175,7 @@
     ).map((chunk) => chunkList(buf2hex(chunk), 2));
   }
 
-  async function calculateRanges(resource, dataLenPromise, colors) {
+  async function calculateRanges(resource, colors) {
```

Comment:
See above comment – this should probably still take dataLength as an input for reactive updates.